File: MM_v1-Django/app/game/static/game/js/utilGame/maxValues.js
// Function responsible for providing the maximum values for the player allows you to display the maximum values for: hull, cargo, masts, crew, cannons, actions,

function maxValues(colour) {

    let xhr = new XMLHttpRequest();
    xhr.onreadystatechange = function() {
        if (xhr.readyState == 4 && xhr.status == 200) {
            let response = JSON.parse(xhr.responseText);
            // let colour = response.playerColour
            console.log(response, colour, 'MAX VALUES')
            console.log(`captain-actions-${response.playerMaxAmountActions}`)

            document.getElementById('player-hit-location-max-value-image').setAttribute('href', response.playerCubeMaxImage);

            // max hull
            if (document.getElementById(`hull-${response.playerMaxAmountHull}-player-hit-location`).getAttribute('href') === "") {
                document.getElementById(`hull-${response.playerMaxAmountHull}-player-hit-location`).setAttribute('href', '#player-hit-location-max-value-image');
            };
            
            // max cargo
            if (document.getElementById(`cargo-${response.playerMaxAmountCargo}-player-hit-location`).getAttribute('href') === "") {
                document.getElementById(`cargo-${response.playerMaxAmountCargo}-player-hit-location`).setAttribute('href', '#player-hit-location-max-value-image');
            };

            // max masts
            if (document.getElementById(`masts-${response.playerMaxAmountMasts}-player-hit-location`).getAttribute('href') === "") {
                document.getElementById(`masts-${response.playerMaxAmountMasts}-player-hit-location`).setAttribute('href', '#player-hit-location-max-value-image');
            };

            // max crew
            if (document.getElementById(`crew-${response.playerMaxAmountCrew}-player-hit-location`).getAttribute('href') === "") {
                document.getElementById(`crew-${response.playerMaxAmountCrew}-player-hit-location`).setAttribute('href', '#player-hit-location-max-value-image');
            };

            // max cannons
            if (document.getElementById(`cannons-${response.playerMaxAmountCannons}-player-hit-location`).getAttribute('href') === "") {
                document.getElementById(`cannons-${response.playerMaxAmountCannons}-player-hit-location`).setAttribute('href', '#player-hit-location-max-value-image');
            };

            // max actions
            document.getElementById('captain-max-value-actions-image').setAttribute('href', response.playerCubeMaxImage);
            document.getElementById(`captain-actions-${response.playerMaxAmountActions}`).setAttribute('href', '#captain-max-value-actions-image');
        };
        
    };
    xhr.open('GET', 'maxValues', true);
    xhr.send();
};

File: MM_v1-Django/app/game/views/maxValues.py
from django.http import JsonResponse
from dataset.utils.dataset.decorators.choices import SHIPSLOCALIZATIONS
from dataset.models import Cube
from game.models import Game
from game.models import TrackPlayerHitLocations
from game.models import PlayersShipsCards

def maxValues(request):
    """
    Maximum parameter values for the player.
    
    Hull, Cargo, Masts, Crew, Cannons, Actions, etc.
    """

    data = {}

    # random colour player active
    game = Game.objects.get(number=100)
    player_colour = request.session['playerColourActive']
    colour_cube_max = Cube.objects.get(name=player_colour.capitalize() + ' Cube Max') # for mark the maximum value of parameter on player board
    colour_cube_max_image = colour_cube_max.image.url
    game_ships = PlayersShipsCards.objects.get(game_number=game)
    player_ship = getattr(game_ships, f"player_{player_colour}")

    # hull
    player_max_hull_ship = player_ship.toughness
    player_max_hull = player_max_hull_ship

    data['playerMaxAmountHull'] = player_max_hull

    # cargo
    player_max_cargo_ship = player_ship.cargo
    player_max_cargo = player_max_cargo_ship
    data['playerMaxAmountCargo'] = player_max_cargo
    
    # masts
    player_max_masts_ship = player_ship.toughness
    player_max_masts = player_max_masts_ship
    data['playerMaxAmountMasts'] = player_max_masts

    # crew
    player_max_crew_ship = player_ship.crew
    player_max_crew = player_max_crew_ship
    data['playerMaxAmountCrew'] = player_max_crew

    # cannons
    player_max_cannons_ship = player_ship.cannons
    player_max_cannons = player_max_cannons_ship
    data['playerMaxAmountCannons'] = player_max_cannons

    # actions
    player_max_speed_ship = player_ship.speed
    player_max_speed = player_max_speed_ship
    data['playerMaxAmountActions'] = player_max_speed

    # data['playerInPort'] = request.session['playerInPort']
    data['playerColour'] = player_colour
    data['playerCubeMaxImage'] = colour_cube_max_image

    return JsonResponse(data)
